A30-xds-v3.md

```diff
@@ -0,0 +1,255 @@
+server, so that it supports only version N.  Before you can do that,
+you need to upgrade the older gRPC clients that speak only versions N-2
+and N-1.  You have two choices:
+
+- Upgrade the clients to a release supporting versions N-1 and N.  These
+  clients will use version N by virtue of the server feature for version
+  N in the bootstrap file.
+- Upgrade the clients to the newest release supporting versions N and
+  N+1.  These clients will not see a server feature for version N+1 in
+  the bootstrap file, so they will use version N.  (They will not
+  actually care whether the server feature for version N is present in
+  the bootstrap file, since they will assume the server supports this
+  version anyway.)
+
+Once you are sure that you have no more clients using version N-1, you
+can drop support for that version from your xDS server.
+
+Next, let's say that xDS version N+1 has been released, and you want to add
+support for that version to your xDS server.  You can of course do this at
+any time, but none of the existing gRPC clients will actually use it
+unless you take additional steps.
+
+- For gRPC clients that support versions N and N+1, all you need to do
+  is add the server feature for version N+1 to the bootstrap file.  That
+  will cause these clients to start using version N+1.
+- For older gRPC clients that support versions N-1 and N, they will
+  ignore the server feature for version N+1 in the bootstrap file, so
+  they will continue to use version N.  You will need to upgrade to a
+  more recent release of gRPC to get them to use version N+1.
+
+Note that the only case in which users will need to upgrade their gRPC
+binaries in lock step with their xDS server is if they are migrating
+their xDS server from version N-1 to version N+1 in a single step,
+without adding support for version N in between.  We do not recommend
+this kind of transition.
+
+### Details of the v2 to v3 Transition
+
+The xDS major version number appears in two main places in the API:
+
+- The RPC method name used to make the xDS request.  For v2, this is
+  `/envoy.service.discovery.v2.AggregatedDiscoveryService/StreamAggregatedResources`,
+  whereas for v3 it is
+  `/envoy.service.discovery.v3.AggregatedDiscoveryService/StreamAggregatedResources`.
+  We call this the *transport protocol version*.
+- The type of resources that we request via the transport protocol.  For
+  example, the type for a Listener resource in v2 is `envoy.api.v2.Listener`,
+  whereas in v3 the type is `envoy.config.listener.v3.Listener`.  We
+  call this the *resource version*.
+
+When gRPC uses v3 to talk to a server, the primary thing that it is
+choosing is the transport protocol version, not the resource version.
+
+Luckily, we can be more flexible with the resource version, because
+there's essentially only one case where we use a field in the v2 protos
+that was removed in the v3 protos.  (That case is the node `build_version`
```

Comment:
Doesn't `build_version` only appear in the transport protocol? I.e. as a part of `DiscoveryRequest`? So for all actual resources, it seems there is no exception (maybe `Node` appears somewhere else?`).